### PR TITLE
State management NgRx

### PR DESCRIPTION
**Pull request by arshithsuresh**

### CODE REVIEW

File: src/app/canvas/canvas.component.ts [hunk 1]
```diff
@@ -16,10 +22,16 @@ export class CanvasComponent implements OnInit, AfterViewInit, OnDestroy {
...
+    constructor(private canvasService: CanvasCoreService, private eventHandlerService: EventHandlerService, private canvasStore: Store) {
         this.shapeCreatedSubscription = this.canvasService.shapeCreated.subscribe((object) => {
             this.drawObject(object);
         });
+
+        // this.canvasStore.dispatch(new AddedObject({
+        //     event: new CanvasEvent(CANVAS_EVENT_TYPE.OBJECT_MODIFIED," "),
+        //     object: new fabric.Rect(),
+        //     canvasState: JSON.stringify(this.canvas)
+        // }));
```
Comment (JohnKuruvila) [inline src/app/canvas/canvas.component.ts]: Remove commented out code.

File: src/app/canvas/canvas.component.ts [hunk 2]
```diff
@@ -33,15 +45,44 @@ export class CanvasComponent implements OnInit, AfterViewInit, OnDestroy {
...
-                    let eventMessage = event.getEventMessage(e);
-                    this.eventHandlerService.handleLogEvent(eventMessage);
+                    let canvasEvent = event.constructEvent(e);
+                    let canvasAction = this.eventHandlerService.handleLogEvent(canvasEvent);
                     event.handleEvent();
+                    // this.canvasStore.dispatch(new AddedObject({
+                    //     event: canvasEvent,
+                    //     object: e.target!,
+                    //     canvasState: JSON.stringify(this.canvas)
+                    // }));
```
Comment (JohnKuruvila) [inline src/app/canvas/canvas.component.ts:72]: Remove commented out code.

File: src/app/state/canvas/canvas.state.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { IDrawableObject } from "src/app/core/objects/object";
+import { fabric } from "fabric";
+export interface ICanvasState {    
+    canvasStates: string[];    
+}
+
+export const initialCanvasState: ICanvasState={
+    canvasStates:[]
```
Comment (JohnKuruvila) [inline src/app/state/canvas/canvas.state.ts]: Store should ideally contain only the present objects on the canvas. Not the history of all the states of the canvas over time. Refer [Three Principles of Redux](https://redux.js.org/understanding/thinking-in-redux/three-principles) to get a few guidelines on how we should use state management.

**Review by JohnKuruvila — CHANGES_REQUESTED** [review]

File: src/app/canvas/canvas.component.ts (hunk 1 above)
Comment (arshithsuresh) [reply src/app/canvas/canvas.component.ts]: removed

File: src/app/canvas/canvas.component.ts (hunk 2 above)
Comment (arshithsuresh) [reply src/app/canvas/canvas.component.ts:72]: removed

File: src/app/state/canvas/canvas.state.ts (hunk 3 above)
Comment (arshithsuresh) [reply src/app/state/canvas/canvas.state.ts]: modified to store only the current state of the canvas.

**Review by JohnKuruvila — APPROVED** [review]

**Pull request merged**